src/main/java/org/ajoberstar/gradle/git/publish/tasks/DirectoryPropertyFactory.java

```diff
@@ -0,0 +1,28 @@
+package org.ajoberstar.gradle.git.publish.tasks;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.file.ProjectLayout;
+import org.gradle.api.model.ObjectFactory;
+
+/**
+ * For backward compatibility with Gradle 4.x
+ */
+public class DirectoryPropertyFactory {
+
+    private DirectoryPropertyFactory() {
+    }
+
+    public static DirectoryProperty create(Project project) {
+        return create(project.getLayout(), project.getObjects());
+    }
+
+    public static DirectoryProperty create(ProjectLayout layout, ObjectFactory objectFactory) {
+        try {
+            return objectFactory.directoryProperty();
+        } catch (NoSuchMethodError e) {
```

Comment:
We could avoid the exception by using `if (GradleVersion.current() < GradleVersion.version('5.0'))` or something similar, right?

Reply:
Right. Would you prefer that?

Reply:
Yeah. If you don't have time, let me know. I plan to make some updates to the plug-in later today.